provisioning/service/src/interfaces.ts

```diff
@@ -18,138 +30,336 @@ export type AttestationTypes = 'none' | 'tpm' | 'x509';
  * Attestation via TPM.
  */
 export interface TpmAttestation {
+  /**
+   * The endorsement key is an encryption key that is permanently embedded in the Trusted Platform Module (TPM)
+   * security hardware, generally at the time of manufacture. This private portion of the endorsement key is never
+   * released outside of the TPM. The public portion of the endorsement key helps to recognize a genuine TPM.
+   *
+   * The endorsement key is a base64 encoded.
+   */
   endorsementKey: string;
+  /**
+   * The storage root key is embedded in the Trusted Platform Module (TPM) security hardware.
+   * It is used to protect TPM keys created by applications, so that these keys cannot be used without the TPM.
+   * Unlike the endorsement key (which is generally created when the TPM is manufactured), the storage root key
+   * is created when you take ownership of the TPM. This means that if you clear the TPM and a new user takes ownership,
+   * a new storage root key is created.
+   *
+   * This property is not typically manipulated by the service client.
+   *
+   * The storageRootKey is a base64 encoded value.
+   */
   storageRootKey: string;
 }
 
 /**
  * Attestation via X509.
  */
 export interface X509Attestation {
+  /**
+   * x509 certificate object which provides all information for a leaf certificate.
+   */
   clientCertificates: X509Certificates;
+  /**
+   * x509 certificate object which provides all information needed for a certificate
+   * suitable for signing other certificates.
+   */
   signingCertificates: X509Certificates;
 }
 
 /**
  * X509 certificate info.
+ *
+ * This object is not provided by the application.  It is populated by the service from
+ * a provided certificate.
  */
 export interface X509CertificateInfo {
+  /**
+   * For clientCertificates the subjectName is used as the deviceId on the IoT Hub.
```

Comment:
>subjectName [](start = 32, length = 11)

same here - also I think we should refer to this as common name ? or CN property of the subject?